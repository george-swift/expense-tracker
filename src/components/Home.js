import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { FaCoins } from 'react-icons/fa';
import { useEffect } from 'react';
import { useVerify } from '../hooks';

const Home = ({ ui }) => {
  const { loggedIn, navigate } = useVerify();

  useEffect(() => {
    if (loggedIn) navigate('et/');
  }, []);

  return (
    <section className="home">
      <div className="brand">
        <h1 className="fw-bold">
          <span className="me-2">Expense Tracker</span>
          <FaCoins />
        </h1>
        <p className="fw-light">Stay in control of your money</p>
      </div>
      <Link to="/signup" className="btn home-btn color-mix" onClick={ui}>Sign Up</Link>
      <Link to="/login" className="btn home-btn" onClick={ui}>Log In</Link>
    </section>
  );
};

Home.propTypes = {
  ui: PropTypes.func.isRequired,
};

export default Home;
