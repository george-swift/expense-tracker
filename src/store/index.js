import { configureStore, combineReducers } from '@reduxjs/toolkit';
import createSagaMiddleware from '@redux-saga/core';
import user from '../slice/user';
import lists from '../slice/lists';
import reports from '../slice/reports';
import expenses from '../slice/expenses';
import notifications from '../slice/notifications';
import rootSaga from '../sagas';

const sagaMiddleware = createSagaMiddleware();

const store = configureStore({
  reducer: combineReducers({
    user,
    lists,
    reports,
    expenses,
    notifications,
  }),
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().prepend(sagaMiddleware),
});

export default store;

sagaMiddleware.run(rootSaga);
