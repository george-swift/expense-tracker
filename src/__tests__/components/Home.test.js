import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, screen, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Home from '../../components/Home';
import store from '../../__mocks__/mockstore';

const ui = jest.fn(() => {});

describe('Snapshot of the Home component', () => {
  it('should render correctly when user visits app', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Home ui={ui} />
          </Router>
        </Provider>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Querying the Home component', () => {
  beforeEach(() => render(
    <Provider store={store}>
      <Router>
        <Home ui={ui} />
      </Router>
    </Provider>,
  ));

  it('should have a button for redirecting new users to sign up', () => {
    const [signUpLink] = screen.getAllByRole('link');

    expect(signUpLink).toBeInTheDocument();

    expect(signUpLink).toHaveAttribute('href', '/signup');

    fireEvent.click(signUpLink);

    expect(ui).toHaveBeenCalledTimes(1);
  });

  it('should have a button for redirecting users to log in', () => {
    const [, logInLink] = screen.getAllByRole('link');

    expect(logInLink).toBeInTheDocument();

    expect(logInLink).toHaveAttribute('href', '/login');

    fireEvent.click(logInLink);

    expect(ui).toHaveBeenCalledTimes(1);
  });
});
