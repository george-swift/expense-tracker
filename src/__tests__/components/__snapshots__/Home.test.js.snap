// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the Home component should render correctly when user visits app 1`] = `
<section
  className="home"
>
  <div
    className="home__banner"
  >
    <h1
      className="fw-bold"
    >
      <span
        className="me-3"
      >
        Expense Tracker
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="CreditScoreTwoToneIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 4H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h5v-2H4v-6h18V6c0-1.11-.89-2-2-2zm0 4H4V6h16v2zm-5.07 11.17-2.83-2.83-1.41 1.41L14.93 22 22 14.93l-1.41-1.41-5.66 5.65z"
        />
      </svg>
    </h1>
    <p
      className="fw-light"
    >
      Stay in control of your money
    </p>
  </div>
  <a
    className="btn home__btn"
    href="/signup"
    onClick={[Function]}
  >
    Sign Up
  </a>
  <a
    className="btn home__btn"
    href="/login"
    onClick={[Function]}
  >
    Log In
  </a>
</section>
`;
